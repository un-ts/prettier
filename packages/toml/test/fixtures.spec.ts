import fs from 'node:fs'
import path from 'node:path'
import { fileURLToPath } from 'node:url'

import { format } from 'prettier'

import TomlPlugin from 'prettier-plugin-toml'

const _dirname =
  typeof __dirname === 'undefined'
    ? path.dirname(fileURLToPath(import.meta.url))
    : __dirname

describe('parser and printer', () => {
  it('should format all fixtures', async () => {
    const fixtures = path.resolve(_dirname, 'fixtures')
    for (const relativeFilepath of fs.readdirSync(fixtures)) {
      const filepath = path.resolve(fixtures, relativeFilepath)
      const input = fs.readFileSync(filepath, 'utf8')

      try {
        const output = await format(input, {
          filepath,
          parser: 'toml',
          plugins: [TomlPlugin],
        })

        expect(output).toMatchSnapshot(relativeFilepath)
      } catch (err: unknown) {
        expect((err as Error).message).toMatchSnapshot(relativeFilepath)
      }
    }
  })
})
