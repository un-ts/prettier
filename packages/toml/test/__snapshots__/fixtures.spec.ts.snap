// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parser and printer > should format all fixtures > comments.toml 1`] = `
"# This is a TOML comment

# This is a multiline
# TOML comment

str1 = \\"I'm a string.\\"
"
`;

exports[`parser and printer > should format all fixtures > fixture1.toml 1`] = `
"[foo]
asfd = 1
bar = [
  'averyloooooooooooooooooooooooooooooooooooooooooooooooooooong',
  \\"arrrrrrrrrrrrrrrrrrrrrrrray\\",
  'thatexceedsthemaximumcolumnwidth',
]
"
`;
